### DISCUSSION TITLE
Different WalletClient objects between `wagmi` and `viem`

### DISCUSSION THREAD

**Discussion by 70nyIT** [wevm/wagmi#3042] [Q&A]
I'm creating an SDK plugin that has hooks accepting a `WalletClient` object to be used with `viem` library methods.

The plugin is intended to be used in react applications using `wagmi`, similarly to 

```typescript
const {data: walletClient} = useWalletClient();
const myHook = useHook(walletClient, otherData);
```

Since `wagmi` and `viem` are very interconnected I was expect that the `viem` [WalletClient object](https://github.com/wagmi-dev/viem/blob/1385d14ccdcc2df1ac76a6de9a9d2476c7ddb7f8/src/clients/createWalletClient.ts#L32) to be the same as `wagmi` [WalletClient object](https://github.com/wagmi-dev/wagmi/blob/1fb735b3ada01eff6944073ff8406d5b0d74fc04/packages/core/src/types/index.ts#L33). Instead I noticed that they're different, causing an error when using the hook.

Specifically, the value `account` in `wagmi` is always of type `Account`, while in `viem` is `Account | undefined`. Is this intended or should be reported as a bug? And if it's intended, why it's different with the same name? Any suggestion on how to correctly use it?

My idea is to import types of `wagm` in my SDK plugin ad mention that I want to accept a `wagmi` `WalletClient` object, still not sure if this will bring other incompatibility. My suggestion would be to keep the same types between `wagmi` and `viem`.

**Comment by 70nyIT** [comment]
I'd like t bump it to see if there's anyone that encountered this error, or if a workaround can be found. I feel that the two `WalletClient` objects should be the same, or at least having different names.

**Comment by AlphaDio** [comment]
Why has this not been answered?

**Comment by AlphaDio** [comment]
```const walletClient = getConnectorClient(config)```

Hope it helps.

**Comment by 70nyIT** [comment]
unfortunately the problem is at TypeScript level, where two objects diff for that optional `account` parameter

**Comment by AlphaDio** [comment]
Yeah, I ended up having to scrap the whole using walletClient for getContract.
